src/Network/Algorand/Node/Api/Content.hs

```diff
@@ -0,0 +1,57 @@
+-- SPDX-FileCopyrightText: 2021 Serokell <https://serokell.io>
+--
+-- SPDX-License-Identifier: MPL-2.0
+
+-- | The REST API v2 of Algod, the Algorand node.
+--
+-- See <https://developer.algorand.org/docs/reference/rest-apis/algod/v2/>
+module Network.Algorand.Node.Api.Content
+  ( Binary
+  , MsgPack
+  , msgPackFormat
+  ) where
+
+import qualified Data.Binary as Binary
+import qualified Data.ByteString.Lazy as BSL
+import qualified Servant.API.ContentTypes as Mime
+
+import Data.ByteString (ByteString)
+import Data.Text (Text)
+import Network.HTTP.Media ((//))
+
+import qualified Data.Algorand.MessagePack as MP
+
+import Data.Algorand.Transaction.Signed (SignedTransaction)
+
+-- | Content type for the `/transactions` endpoint.
+data Binary
+
+instance Mime.Accept Binary where
+  contentType _ = "application" // "x-binary"
+
+instance Mime.MimeRender Binary ByteString where
+  mimeRender _ = BSL.fromStrict
+
+instance Mime.MimeRender Binary BSL.ByteString where
+  mimeRender _ = id
+
+instance Mime.MimeRender Binary [SignedTransaction] where
+  mimeRender _ = mconcat . map (MP.pack . MP.Canonical)
+
+-- | Content type for the endpoints which can return msgpack.
+data MsgPack
+
+msgPackFormat :: Maybe Text
+msgPackFormat = Just "msgpack"
```

Comment:
Now that it stays in a different module (outside of the place where API is defined), I think it should be commented. Let's say that it can be passed as `format` parameter.